openshift/process_template.go

```diff
@@ -134,7 +136,12 @@ func LoadProcessedTemplates(ctx context.Context, config Config, username string,
 		token := goajwt.ContextJWT(ctx)
 		if token != nil {
 			vars["OSIO_TOKEN"] = token.Raw
+			id := token.Claims.(jwt.MapClaims)["sub"]
+			if id != nil {
```

Comment:
If there is no "sub"/ID in the token then there is something fundamentally broken... You should return an error rather than silently ignore it.

Reply:
in this IMO it would be better to reuse existing API (we can leave with double token extraction)

Reply:
@ibuziuk 

I can do it if you like, but then I'll have to check for the text of the error, to throw an error if `sub` is not there, but *not throw an error* when the token itself is not there ? We shouldn't throw an error if the token itself is not there.

Seems to me that this would make the code even more complicated, no ?

FTR the code of the function I would reuse is:

```
func extractIdentityID(ctx context.Context) (string, error) {
	token := goajwt.ContextJWT(ctx)
	if token == nil {
		return "", errors.New("Missing token")
	}
	id := token.Claims.(jwt.MapClaims)["sub"]
	if id == nil {
		return "", errors.New("Missing sub")
	}

	return id.(string), nil
}
```

Reply:
@alexeykazakov 

> If there is no "sub"/ID in the token then there is something fundamentally broken... You should return an error rather than silently ignore it.

I'm not sure it's the place to make such a check and throw such an error, but if you think it is I can throw an error when `id == nil`. Should I throw an exception ? FYI I'm a Go newbie, so if you could provide me with the piece of code you would add to correctly manage this error, that would be great :-)

Reply:
`to throw an error if sub is not there, but not throw an error when the token itself is not there ? We shouldn't throw an error if the token itself is not there.`

is it possible scenario when context is missing token, or it also means that something is broken ?

Reply:
I assume that if the tenant update if run by some sort of batch command (for a whole group of users for example), then the Keycloak token coming from the user http request will not be there.

Reply:
ok, whatever works than ;-)

Reply:
> If there is no "sub"/ID in the token then there is something fundamentally broken... You should return an error rather than silently ignore it.

I now return an error in this case. Thanks @xcoulon for your help.